src/java/org/apache/cassandra/db/commitlog/CommitLogReplayer.java

```diff
@@ -276,15 +283,16 @@ public void runMayThrow()
                             if (newPUCollector == null)
                                 newPUCollector = new Mutation.PartitionUpdateCollector(mutation.getKeyspaceName(), mutation.key());
                             newPUCollector.add(update);
-                            commitLogReplayer.replayedCount.incrementAndGet();
+                            replayedCount++;
                         }
                     }
                     if (newPUCollector != null)
                     {
                         assert !newPUCollector.isEmpty();
 
                         Keyspace.open(newPUCollector.getKeyspaceName()).apply(newPUCollector.build(), false, true, false);
-                        commitLogReplayer.keyspacesReplayed.add(keyspace);
+                        commitLogReplayer.keyspacesReplayed.computeIfAbsent(keyspace, k -> new AtomicInteger(0))
```

Comment:
what if `Keyspace.open` or something else in the loop throw? Are we ok with the counter being off?

Reply:
If it throws, entire replay will be aborted in `blockForWrites`. The accumulated `keyspacesReplayed` won't be used again